### PR TITLE
maint: Added base64 gem explicitly dependency

### PR DESCRIPTION
**Pull request by hsbt**
### Before the change?

octokit.rb warns the following message With Ruby 3.3.0.

```
/Users/hsbt/.local/share/gem/gems/octokit-8.0.0/lib/octokit/client/code_scanning.rb:3: warning: base64 was loaded from the standard library, but is not part of the default gems since Ruby 3.4.0. Add base64 to your Gemfile or gemspec. Also contact author of octokit-8.0.0 to add base64 into its gemspec.
```

This means octokit.rb is not working after Ruby 3.4 release. We should add `base64` as dependency.

### After the change?

Above warning is suppressed.

### Pull request checklist
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been reviewed and added / updated if needed (for bug fixes / features)

### Does this introduce a breaking change?

Please see our docs on [breaking changes](https://github.com/octokit/.github/blob/master/community/breaking_changes.md) to help!

- [ ] Yes
- [x] No

----

### CODE REVIEW

**Review by nickfloyd — CHANGES_REQUESTED** [review]
Hey @hsbt Thanks for the contributions here!  Before we can move forward would you mind addressing the lint issue of making sure the deps are in the spec file in alphabetical order.  That way CI can run complete.

Alternatively you can give owner commit access to your fork and I can make the changes.

**Comment by hsbt** [comment]
@nickfloyd done to sort alphabetical order.

**Review by xuanxu — APPROVED** [review]
:+1:

File: octokit.gemspec
Comment (nickfloyd) [inline octokit.gemspec:8]:
```suggestion
  spec.add_dependency 'base64' if Gem::Version.new(RUBY_VERSION) >= Gem::Version.new('3.3')
```

What are thoughts about approaching it like this? I know that it shouldn't matter but I would like to reduce the surface area of the impact.  Given that pre v3.3 can get`base64` from stdlib without the warn I think this approach should be ok.

Let me know your thoughts when you get the chance!  Thanks.

**Review by nickfloyd — COMMENTED** [review]
Let me know your thoughts on my suggestion around reducing the scope of impact.

**Review by nickfloyd — APPROVED** [review]
I am going to favor getting this in given our 3.4 builds/ci are failing - we can discuss the merits of making it a conditional dependency later. 

Thanks @hsbt for the contributions here ❤️ !

File: octokit.gemspec
Comment (hsbt) [reply octokit.gemspec:8]: It's not working. I'm member of Ruby core team and RubyGems team.

RubyGems couldn't handle condition of Ruby version for dependency. RubyGems only support `required_ruby_version` and `required_rubygems_version` for gemspec, not dependency.

Comment (nickfloyd) [reply octokit.gemspec:8]: Thanks for the clarity!  I've merged it as is and will ship it once I get a few more changesets in.

**Comment by hsbt** [comment]
Thanks!

**Pull request merged**
